spec/features/user_can_correct_submission_with_no_business_spec.rb

```diff
@@ -0,0 +1,111 @@
+require 'rails_helper'
+
+RSpec.feature 'Correcting a submission by reporting no business' do
+  before do
+    mock_sso_with(email: 'email@example.com')
+    mock_incomplete_tasks_endpoint!
+    mock_complete_tasks_endpoint!
+    mock_task_with_framework_endpoint!
+    mock_no_business_correction_endpoint!
+    mock_submission_completed_no_business_endpoint!
+  end
+
+  context 'when the latest submission is a business return' do
+    before do
+      mock_completed_task_endpoint!
+    end
+
+    scenario 'user corrects a submission, reporting "no business" when linked to one supplier' do
+      cached_correction_returns_enabled = ENV['CORRECTION_RETURNS_ENABLED']
+      ENV['CORRECTION_RETURNS_ENABLED'] = 'yes'
+
+      mock_user_endpoint!
+
+      visit '/'
+      click_link 'sign-in'
+
+      click_link 'History'
+      click_link 'View'
+      click_link 'Correct this return'
+
+      expect(page).to have_content 'Cheese Board 5'
+      expect(page).to have_content 'July 2018'
+      expect(page).to_not have_content 'Bobs Cheese Shop'
+
+      click_on 'Report no business'
+
+      expect(page).to have_content 'This is a correction'
+      expect(page).to have_content 'Report no business for CBOARD5'
+      expect(page).to_not have_content 'Bobs Cheese Shop'
+
+      click_on 'Confirm no business'
+
+      expect(page).to have_content 'This is a correction'
+      expect(page).to have_content 'CBOARD5'
+      expect(page).to have_content 'July 2018'
+      expect(page).to have_content 'Thank you for reporting no business for'
+      expect(page).to_not have_content 'Bobs Cheese Shop'
+
+      ENV['CORRECTION_RETURNS_ENABLED'] = cached_correction_returns_enabled
```

Comment:
If the spec fails above then this line will never get evaluated, leaving the env var value set to 'yes' for the rest of the test run, which could cause unrelated/red-herring failures elsewhere. The normal way to avoid this problem is to either memoise and reset the value in `before`/`after` blocks so that they are always evaluated, or by using the [`ensure`](https://ruby-doc.org/docs/ruby-doc-bundle/UsersGuide/rg/ensure.html) keyword just before this line so that the reset is always evaluated.